third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -850,28 +839,23 @@ impl AccessSpecifier {
 )]
 pub enum AccessKind {
     Reads,
-    Writes,
-    Acquires, // reads or writes
+    Writes, // Includes reading
 }
 
 impl AccessKind {
     /// Returns true if this access kind subsumes the other.
     pub fn subsumes(&self, other: &Self) -> bool {
         use AccessKind::*;
-        match (self, other) {
-            (Acquires, _) => true,
-            (_, Acquires) => false,
-            _ => self == other,
-        }
+        matches!((self, other), (_, Reads) | (Writes, Writes))
     }
 
     /// Tries to join two kinds, returns None if no intersection.
     pub fn try_join(self, other: Self) -> Option<Self> {
         use AccessKind::*;
-        match (self, other) {
-            (Acquires, k) | (k, Acquires) => Some(k),
-            (k1, k2) if k1 == k2 => Some(k1),
-            _ => None,
+        if matches!((self, other), (Writes, Writes)) {
+            Some(self)
+        } else {
+            Some(Reads)
```

Comment:
Read and write are treated as a read because of read-before-write? This behaviour is different from before, also you should change the doc string and maybe even function signature (it is always Some?)